lib/model/model.go

```diff
@@ -1151,46 +1278,94 @@ func (m *model) ClusterConfig(deviceID protocol.DeviceID, cm protocol.ClusterCon
 		m.Add(is)
 	}
 
-	m.pmut.Lock()
-	m.remotePausedFolders[deviceID] = paused
-	m.pmut.Unlock()
+	return changed, tempIndexFolders, paused, nil
+}
 
-	// This breaks if we send multiple CM messages during the same connection.
-	if len(tempIndexFolders) > 0 {
-		m.pmut.RLock()
-		conn, ok := m.conn[deviceID]
-		m.pmut.RUnlock()
-		// In case we've got ClusterConfig, and the connection disappeared
-		// from infront of our nose.
-		if ok {
-			m.progressEmitter.temporaryIndexSubscribe(conn, tempIndexFolders)
-		}
+func (m *model) ccCheckEncryption(fcfg config.FolderConfiguration, folderDevice config.FolderDeviceConfiguration, ccDeviceRemote, ccDeviceLocal protocol.Device, hasDeviceRemote, hasDeviceLocal, deviceUntrusted bool) error {
+	hasTokenRemote := hasDeviceRemote && len(ccDeviceRemote.EncPwToken) > 0
+	hasTokenLocal := hasDeviceLocal && len(ccDeviceLocal.EncPwToken) > 0
+	isEncRemote := folderDevice.EncryptionPassword != ""
+	isEncLocal := fcfg.Type == config.FolderTypeReceiveEncrypted
+
+	if !isEncRemote && !isEncLocal && deviceUntrusted {
```

Comment:
This transience of untrustedness for encrypted folders is weird.
It seems if I have
A (plain) -> B (encrypted) -> C (encrypted), B still requires C to be marked as untrusted?

Reply:
B requires that either 1. like in the cc from C, C marks itself as encrypted, or 2., like in the cc from A, A is not marked as encrypted, but B is. Also relevant:
https://github.com/syncthing/syncthing/pull/6727#discussion_r484730030

Reply:
So I need to think more about this, but something smells to me here.